src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -57,5 +83,142 @@ public void SendAsync_SimpleGet_Success()
                     "Expected cancellation requested on token attached to exception.");
             }
         }
+
+        [Fact]
+        public void GetAsyncAutomaticDecompression()
+        {
+            var handler = new HttpClientHandler();
+            if (!handler.SupportsAutomaticDecompression)
+            {
+                return;
+            }
+
+            var client = new HttpClient(handler);
+            var response = client.GetAsync(HttpTestServers.RemoteServerGzipUri).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var responseContent = response.Content.ReadAsStringAsync().Result;
+            Assert.True(responseContent.Contains(DecompressedContentPart));
+            handler.Dispose();
+            client.Dispose();
+        }
+
+        [Fact]
+        public void GetAsyncBasicAuthentication()
+        {
+            var uri = HttpTestServers.BasicAuthUriForCreds(Username, Password);
+            var unAuthClient = new HttpClient();
+            var unAuthResponse = unAuthClient.GetAsync(uri).Result;
+            Assert.Equal(HttpStatusCode.Unauthorized, unAuthResponse.StatusCode);
+            unAuthClient.Dispose();
+
+            var handler = new HttpClientHandler();
+            handler.Credentials = _credential;
+            var client = new HttpClient(handler);
+            var response = client.GetAsync(uri).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(new Version(1,1), response.Version);
+            handler.Dispose();
+            client.Dispose();
+        }
+
```

Comment:
As further comments on this test design, there are too many things being tested in this one 'test'.

This test, for example, should be split up into at least 2 tests with names something like this:

GetAsync_UseNoAuthentication_Unauthorized()
GetAsync_UseBasicAuthentication_OK()

etc.

In general, tests should conform to the "Arrange, Act, Assert" principle.